syntax = "proto3";
import "common.proto";
package dasproto;

message MessageData {
    string command = 1;
    repeated string args = 2;
    string sender = 3;
    bool is_broadcast = 4;
    repeated string visited_recipients = 5;
}

service AtomSpaceNode {
    rpc ping (Empty) returns (Ack) {}
    rpc execute_message (MessageData) returns (Empty) {}
}
